cypress/e2e/registration.cy.js

```diff
@@ -1,11 +1,84 @@
 /// <reference types='cypress' />
 
 describe('Student Registration page', () => {
-  before(() => {
+  let user;
 
+  before(() => {
+    cy.visit('/');
+    cy.task('generateUser').then((generateUser) => {
+      user = generateUser;
+    });
   });
 
-  it('', () => {
+  it('should register a new student', () => {
+    cy.findByPlaceholder('First Name')
+      .type(user.firstName);
+
+    cy.findByPlaceholder('Last Name')
+      .type(user.lastName);
+
+    cy.get('#userEmail').type(user.email);
+
+    cy.contains('.custom-control-label', user.gender)
+      .click();
+
+    cy.findByPlaceholder('Mobile Number')
+      .type(user.mobileNumber);
+
+    cy.get('#dateOfBirthInput')
+      .type('{selectAll}15 Sep 2005{enter}');
+
+    cy.get('.subjects-auto-complete__value-container')
+      .type('m{enter}' + 'e{enter}' + 'p{enter}');
```

Comment:
The way the subjects are selected is not clear and hard to understand. Consider using a more explicit method to select multiple options in a dropdown.